### PR TITLE
simplify create-svelte

### PR DESCRIPTION
**Pull request by Rich-Harris**
#987

### Before submitting the PR, please make sure you do the following
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [ ] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts

### CODE REVIEW

**Comment by Rich-Harris** [comment]
I think this is a bit simpler — no CSS/SASS/LESS option, and the template (soon to be _templates_) is authored in TS and converted to JS at build time.

Additionally, the modifiers no longer make brittle edits via code; instead we have files inside directories like

```
shared/+eslint-prettier
```

which means 'copy this if the user chose eslint but not prettier, regardless of their typescript choice'. It results in some duplication, but I think it'll prove easier to grok.

Also, since more of the work is happening at build time, the `create-svelte` package gets even tinier.

Not sure what's going on with the lint failure, will investigation in the morning.

**Comment by benmccann** [comment]
FYI, `lint` is failing

The naming scheme seems like it could create an exponential explosion. E.g. we have four eslint configs right now. If we add a prompt for legacy mode (i.e. IE support) that used `@babel/eslint-plugin` I think we would need to create a `+babel` and `-babel` version (or `+legacy`/`-legacy` if we want to call it that instead) of each of the existing files so we'd have eight of them.

File: packages/create-svelte/rollup.config.js [hunk 1]
```diff
@@ -5,8 +5,8 @@ import json from '@rollup/plugin-json';
 export default {
 	input: 'cli/index.js',
 	output: {
-		file: 'bin',
-		format: 'cjs',
+		file: 'bin.js',
+		format: 'esm',
```
Comment (dummdidumm) [inline packages/create-svelte/rollup.config.js:9]: Why change the format? Is this even related to the changes in this PR?

File: packages/create-svelte/templates/default/jsconfig.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+	"compilerOptions": {
+		"baseUrl": ".",
+		"paths": {
+			"$app/*": [".svelte/dev/runtime/app/*", ".svelte/build/runtime/app/*"],
```
Comment (dummdidumm) [inline packages/create-svelte/templates/default/jsconfig.json:5]: Should be removed, there are ambient module type definitions now 
```suggestion
```

File: packages/create-svelte/rollup.config.js (hunk 1 above)
Comment (Rich-Harris) [reply packages/create-svelte/rollup.config.js:9]: there's just no reason to use CJS 🤷

Comment (dummdidumm) [reply packages/create-svelte/rollup.config.js:9]: That's true, it just threw me off a little that this change is part of the PR 😄

File: packages/create-svelte/scripts/build-templates.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
+import fs from 'fs';
+import path from 'path';
+import parser from 'gitignore-parser';
+import prettier from 'prettier';
+import { transform } from 'sucrase';
```
Comment (dummdidumm) [inline packages/create-svelte/scripts/build-templates.js:5]: What's the reason to use `sucrase` here? If only TS is transpiled, we don't need it and can use `ts.transpileModule` instead.

**Comment by Rich-Harris** [comment]
I hear the point re combinatorial explosion, but my sense is that it's less of an issue than it appears. Firstly, the package.json files are merged (i.e. the `eslint` and `eslint-plugin-svelte3` dev dependencies are only declared in the `+eslint` directory, not `+eslint` and `+eslint-prettier` and `+eslint-typescript` and `+eslint+prettier` and `+eslint+prettier-typescript` and `+eslint+prettier+typescript` and `+eslint+typescript`).

I can't imagine us wanting to add a Babel option other than to support a `legacy` mode, in which case we wouldn't need `@babel/eslint-plugin` IIUC — that's only to extend the ESLint parser with support for new language features. To my mind that falls very much under 'if you want to add a bunch of complexity to your project, go nuts, but you're on your own'.

In cases where tools _don't_ overlap (i.e. adding Babel _without_ adding the ESLint plugin), no combinatorial explosion will arise; it's just like that right now because TypeScript and Prettier and ESLint _do_ overlap. It certainly could become an issue in future but I think we can probably cross that bridge when we come to it. I still think on balance it's better to have things be declarative even at the cost of some duplication. It's definitely not clear-cut though.

File: packages/create-svelte/scripts/build-templates.js (hunk 3 above)
Comment (Rich-Harris) [reply packages/create-svelte/scripts/build-templates.js:5]: Sucrase preserves the original formatting better. If you use TS directly, this...

```svelte
<script lang="ts">
	let count = 0;

	const increment = () => {
		count += 1;
	};
</script>
```

...gets turned into this:

```svelte
<script>
	let count = 0;
	const increment = () => {
		count += 1;
	};
</script>
```

It's a little thing but it matters

Comment (dummdidumm) [reply packages/create-svelte/scripts/build-templates.js:5]: Didn't know that, makes sense. Maybe we should add a code comment then for future devs who think the same as me did first.

**Comment by dummdidumm** [comment]
Yeah I'd also say if we add a bunch more stuff and there are too many combinations, we can always revisit this decision. So I'm good with these changes.

File: packages/create-svelte/scripts/build-templates.js (hunk 3 above)
Comment (Rich-Harris) [reply packages/create-svelte/scripts/build-templates.js:5]: good point, have done that

File: packages/create-svelte/shared/+typescript/svelte.config.cjs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+module.exports = {
+	kit: {
+		// By default, `npm run build` will create a standard Node app.
+		// You can create optimized builds for different platforms by
+		// specifying a different adapter
+		adapter: node(),
+
+		// Consult https://github.com/sveltejs/svelte-preprocess
+		// for more information about preprocessors
+		preprocess: sveltePreprocess(),
```
Comment (ignatiusmb) [inline packages/create-svelte/shared/+typescript/svelte.config.cjs:15]: Shouldn't this be at the same level as `kit`?

Comment (Rich-Harris) [reply packages/create-svelte/shared/+typescript/svelte.config.cjs:15]: good catch, thanks — fixed in 3c41d0756678efe96c887df80205a00816845a0e
